### PR TITLE
test(compactor): Add test for large amounts of data with a single timestamp

### PR DESCRIPTION
**Pull request by alamb**
During the develpment of https://github.com/influxdata/influxdb_iox/pull/7079 @NGA-TRAN , @Joe-Blount  and I came across a corner case for the compactor if there is lots of with a single timestamp. While this is not the expected use case, it could happen either if 
1. Someone made a mistake and wrote in the same timestamp for a large number of rows (more likely)
2. Someone was stress testing the system

The compactor doesn't necessarily have to handle actually splitting such files, but it shouldn't crash/ panic / etc either.

Add tests that cover these cases

### CODE REVIEW

File: compactor2/tests/layouts/single_timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    - "**** Input Files "
+    - "L0, all files 4.88gb                                                                                "
+    - "L0.1[100,100]       |-------------------------------------L0.1-------------------------------------|"
+    - "Committing partition 1:"
+    - "  Upgrading 1 files level to CompactionLevel::L1: L0.1"
+    - "Committing partition 1:"
+    - "  Upgrading 1 files level to CompactionLevel::L2: L1.1"
+    - "**** Final Output Files "
+    - "L2, all files 4.88gb                                                                                "
+    - "L2.1[100,100]       |-------------------------------------L2.1-------------------------------------|"
```
Comment (alamb) [inline compactor2/tests/layouts/single_timestamp.rs:44]: This is kind if neat that it handles this case very well and just upgrades the file

File: compactor2/tests/layouts/single_timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+    insta::assert_yaml_snapshot!(
+        run_layout_scenario(&setup).await,
+        @r###"
+    ---
+    - "**** Input Files "
+    - "L0, all files 4.88gb                                                                                "
+    - "L0.1[100,100]       |-------------------------------------L0.1-------------------------------------|"
+    - "L0.2[100,100]       |-------------------------------------L0.2-------------------------------------|"
+    - "SKIPPED COMPACTION for PartitionId(1): partition 1 has 10485760000 parquet file bytes, limit is 268435456"
```
Comment (alamb) [inline compactor2/tests/layouts/single_timestamp.rs:78]: this scenario is skipped as expected

File: compactor2_test_utils/src/simulator.rs [hunk 3]
```diff
@@ -348,5 +348,25 @@ fn even_time_split(
                 compaction_level,
             }
         })
-        .collect()
+        .collect();
+
+    // make sure we have assigned all bytes and rows to one of the output files
+    if simulated_files.len() == 1 {
```
Comment (alamb) [inline compactor2_test_utils/src/simulator.rs:354]: I will create a follow on PR apply this fix to all cases (where the rounding leads to losing rows / data). In the real world, I think the output files would actually likely be slightly smaller, though the row counts shouldn't change)

File: compactor2/tests/layouts/single_timestamp.rs [hunk 4]
```diff
@@ -0,0 +1,244 @@
+//! layout tests for the pathalogical cases of large amounts of data
+//! with a single timestamp.
+//!
+//! The compactor doens't necessarily have to handle actually
```
Comment (Joe-Blount) [inline compactor2/tests/layouts/single_timestamp.rs:4]:
```suggestion
//! The compactor doesn't necessarily have to handle actually
```

File: compactor2_test_utils/src/simulator.rs (hunk 3 above)
Comment (alamb) [reply compactor2_test_utils/src/simulator.rs:354]: https://github.com/influxdata/influxdb_iox/pull/7124

**Review by NGA-TRAN — APPROVED** [review]
Nice.Thanks @alamb

**Review by Joe-Blount — APPROVED** [review]

File: compactor2/tests/layouts/single_timestamp.rs (hunk 2 above)
Comment (NGA-TRAN) [reply compactor2/tests/layouts/single_timestamp.rs:78]: This is the behavior of #7079 
`panicked at 'Not enough memory budget to compact the minimum set of files (1 start-level file with at most 1 target-level file). Please adjust compactor settings', compactor2/src/components/split_or_compact/split_compact.rs:64:17`

I will modify the code to skip it

**Pull request merged**